libsolidity/formal/SMTEncoder.cpp

```diff
@@ -877,6 +877,27 @@ bool SMTEncoder::visit(MemberAccess const& _memberAccess)
 			return false;
 		}
 	}
+	else if (exprType->category() == Type::Category::Array)
+	{
+		_memberAccess.expression().accept(*this);
+		if (_memberAccess.memberName() == "length")
+		{
+			auto symbArray = dynamic_pointer_cast<smt::SymbolicArrayVariable>(m_context.expression(_memberAccess.expression()));
+			solAssert(symbArray, "");
+			defineExpr(_memberAccess, symbArray->length());
+			m_uninterpretedTerms.insert(&_memberAccess);
+		}
+		else
+		{
+			auto const& name = _memberAccess.memberName();
+			solAssert(name == "push" || name == "pop", "");
```

Comment:
Can't you trigger this with the following?
```
library L { function f(uint256[] memory) public pure {} }
contract C { using L for uint256[]; function g() public pure { uint256[] memory x; x.f(); } }
```
Or something like that?

Reply:
It wasn't triggered, seems fine, but I added an assertion that didn't work...

Reply:
Ah, I now see it - the ``if (accessType->category() == Type::Category::Function)`` above catches these I think, so it's fine!

Reply:
Although, it might be worth to check/verify that this either actually works or produces a warning, but that we can do later.

Reply:
I'd actually suggest removing the `else` completely, since https://github.com/ethereum/solidity/pull/8916 removes it anyway

Reply:
Well keeping it is fine as well and having the unsupported warning until #8916 is merged doesn't hurt, does it? But I'll also reapprove, if you remove it.